cmd/vpu_plugin/vpu_plugin.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/google/gousb"
+	"os"
+	"time"
+
+	"github.com/intel/intel-device-plugins-for-kubernetes/pkg/debug"
+	dpapi "github.com/intel/intel-device-plugins-for-kubernetes/pkg/deviceplugin"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	// Movidius MyriadX Vendor ID
+	vendorID = 0x03e7
+	// Device plugin settings.
+	namespace  = "vpu.intel.com"
+	deviceType = "hddl"
+
+	hddlSockPath     = "/var/tmp/hddl_service.sock"
+	hddlServicePath1 = "/var/tmp/hddl_service_ready.mutex"
+	hddlServicePath2 = "/var/tmp/hddl_service_alive.mutex"
+	ionDevNode       = "/dev/ion"
+)
+
+var (
+	isdebug = flag.Int("debug", 0, "debug level (0..3)")
+	// Movidius MyriadX Product IDs
+	productIDs = []int{0x2485, 0xf63b}
+)
+
+type gousbContext interface {
+	OpenDevices(opener func(desc *gousb.DeviceDesc) bool) ([]*gousb.Device, error)
+}
+
+type devicePlugin struct {
+	usbContext   gousbContext
+	vendorID     int
+	productIDs   []int
+	sharedDevNum int
+}
+
+func newDevicePlugin(usbContext gousbContext, vendorID int, productIDs []int, sharedDevNum int) *devicePlugin {
+	return &devicePlugin{
+		usbContext:   usbContext,
+		vendorID:     vendorID,
+		productIDs:   productIDs,
+		sharedDevNum: sharedDevNum,
+	}
+}
+
+func (dp *devicePlugin) Scan(notifier dpapi.Notifier) error {
+	for {
+		devTree, err := dp.scan()
+		if err != nil {
+			return err
+		}
+
+		notifier.Notify(devTree)
+
+		time.Sleep(5 * time.Second)
```

Comment:
Not a query specifically about this PR per-se, as this code I think appears in all the plugins - but, what's with the 5s delay spin/poll loop? Is that a specific design feature? We might want to document that somewhere...

Reply:
It first appeared in the GPU plugin. Since the i915 driver didn't support any notification mechanism for hot plugged devices at that time the simplest possible poll loop was used (and copy-pasted to other plugins). It could be a `for` loop over a channel receiving e.g. inotify events.